From e65f3041b1a14dc7d46935091b35ae6a7236d118 Mon Sep 17 00:00:00 2001
From: Michal Vasilek <michal@vasilek.cz>
Date: Sun, 18 Sep 2022 19:32:51 +0200
Subject: [PATCH] Fix fallback metric unit detection logic

When HAVE__NL_MEASUREMENT_MEASUREMENT is not defined (for example on
musl systems), the fallback logic checks for units in translation files.

If the unit in translation files is metric, we should use metric and
not print a warning about missing translation and use metric.

Introduced in 1c140fc8ce08260d5008847945bf345654ad7fa8
---
 libgweather/gweather-info.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libgweather/gweather-info.c b/libgweather/gweather-info.c
index d33d3905..193ee86c 100644
--- a/libgweather/gweather-info.c
+++ b/libgweather/gweather-info.c
@@ -880,7 +880,7 @@ is_locale_metric (void)
 
     if (strcmp (e, "default:inch") == 0)
         return FALSE;
-    else if (strcmp (e, "default:mm") == 1)
+    else if (strcmp (e, "default:mm") == 0)
         return TRUE;
     else {
         g_warning ("Wrong translation for libgweather; please file "
-- 
2.37.3

